WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationActivity.kt

```diff
@@ -110,52 +113,47 @@ class SiteCreationActivity : LocaleAwareActivity(),
                         Triple(true, null, createSiteState.isSiteTitleTaskComplete)
                     }
                     is SiteCreationCompleted -> Triple(
-                            true, createSiteState.localSiteId,
-                            createSiteState.isSiteTitleTaskComplete
+                        true, createSiteState.localSiteId,
+                        createSiteState.isSiteTitleTaskComplete
                     )
                 }
                 intent.putExtra(SitePickerActivity.KEY_SITE_LOCAL_ID, localSiteId)
                 intent.putExtra(SitePickerActivity.KEY_SITE_TITLE_TASK_COMPLETED, titleTaskComplete)
                 setResult(if (siteCreated) Activity.RESULT_OK else Activity.RESULT_CANCELED, intent)
                 finish()
             }
-        })
-        mainViewModel.dialogActionObservable.observe(this, Observer { dialogHolder ->
-            dialogHolder?.let {
-                val supportFragmentManager = requireNotNull(supportFragmentManager) {
-                    "FragmentManager can't be null at this point"
-                }
-                dialogHolder.show(this, supportFragmentManager, uiHelpers)
-            }
-        })
-        mainViewModel.exitFlowObservable.observe(this, Observer {
+        }
+        mainViewModel.dialogActionObservable.observe(this) {
+            it?.show(this, supportFragmentManager, uiHelpers)
+        }
+        mainViewModel.exitFlowObservable.observe(this) {
             setResult(Activity.RESULT_CANCELED)
             finish()
-        })
-        mainViewModel.onBackPressedObservable.observe(this, Observer {
+        }
+        mainViewModel.onBackPressedObservable.observe(this) {
             ActivityUtils.hideKeyboard(this)
-            super.onBackPressed()
-        })
-        siteCreationIntentsViewModel.onBackButtonPressed.observe(this, Observer {
+            onBackPressedDispatcher.onBackPressed()
```

Comment:
Question (❓): This `onBackPressedDispatcher.onBackPressed()` is replacing a `super.onBackPressed()` call and not a normal `onBackPressed()` call, is that intended here? 🤔

Reply:
Another good catch! The behavior was also broken here.
Fixed: 3a3110e41098bce85f6d8cfd900ef481c7801279

Reply:
Awesome, thanks for the fix @irfano ! 🥇 

Btw, this is the only place do also `backPressedCallback.isEnabled = true` afterwards, was there a reason for that? 🤔

Reply:
> Btw, this is the only place do also backPressedCallback.isEnabled = true afterwards, was there a reason for that? 🤔

`super.onBackPressed()` closes the fragment while `SiteCreationActivity`  is still alive. That's why we still need the callback to be enabled after calling `onBackPressedDispatcher.onBackPressed()`. This is an example of the concern I shared in the [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/18019#discussion_r1123718077). 
Other classes don't have the same behavior, they just finish the activity on `super`onBackPressed()`. But still, to be on the safe side, I have decided to do the same everywhere, not just here. Additionally, I will be using an extension function for better manageability.

Reply:
Great, thanks for the explanation and for applying the same behavior everywhere else @irfano ! 🚀